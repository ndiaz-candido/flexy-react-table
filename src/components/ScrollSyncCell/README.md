This component is aim to be just for a declarative purpose.
It should be used to indicate the column where its children should be rendered.

```js
<ScrollSyncTable stickHeader>
  <ScrollSyncColumns>
    <ScrollSyncColumn name="a" width="100px">Header A</ScrollSyncColumn>
    <ScrollSyncColumn name="b" width="100px">Header B</ScrollSyncColumn>
  </ScrollSyncColumns>
  <ScrollSyncRows>
    <ScrollSyncRow>
      <ScrollSyncCell column="a">Column A</ScrollSyncCell>
      <ScrollSyncCell column="b">Column B</ScrollSyncCell>
    </ScrollSyncRow>
  </ScrollSyncRows>
</ScrollSyncTable>
```
