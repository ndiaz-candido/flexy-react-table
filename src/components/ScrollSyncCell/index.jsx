import React from 'react';
import PropTypes from 'prop-types';

const ScrollSyncCell = ({ column }) => (<div>Cell rendering content for { column }</div>);

ScrollSyncCell.propTypes = {
  /** The column name where the cell should be rendered */
  column: PropTypes.string.isRequired
};

export default ScrollSyncCell;
