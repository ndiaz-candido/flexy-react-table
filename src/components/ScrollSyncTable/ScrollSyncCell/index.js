import ScrollSyncCell from './ScrollSyncCell';

export default ScrollSyncCell;
