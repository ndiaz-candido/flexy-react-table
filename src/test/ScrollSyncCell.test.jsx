import React from 'react';
import { shallow } from 'enzyme';
import ScrollSyncCell from '../components/ScrollSyncCell';

describe('ScrollSyncCell', () => {
  test('should render its default content', () => {
    const ScrollSyncCellComponent = shallow(<ScrollSyncCell column="a" />);

    expect(ScrollSyncCellComponent).toMatchSnapshot();
  });
});
