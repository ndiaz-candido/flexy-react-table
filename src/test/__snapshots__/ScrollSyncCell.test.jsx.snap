// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollSyncCell should render its default content 1`] = `""`;
